base/docker-integration.go

```diff
@@ -0,0 +1,320 @@
+				"8095/tcp": {{"0.0.0.0", "8095"}},
+				"8096/tcp": {{"0.0.0.0", "8096"}},
+
+				"11207/tcp": {{"0.0.0.0", "11207"}},
+				"11210/tcp": {{"0.0.0.0", "11210"}},
+				"11211/tcp": {{"0.0.0.0", "11211"}},
+
+				"18091/tcp": {{"0.0.0.0", "18091"}},
+				"18092/tcp": {{"0.0.0.0", "18092"}},
+				"18093/tcp": {{"0.0.0.0", "18093"}},
+				"18094/tcp": {{"0.0.0.0", "18094"}},
+				"18095/tcp": {{"0.0.0.0", "18095"}},
+				"18096/tcp": {{"0.0.0.0", "18096"}},
+			}
+		})
+		if err != nil {
+			log.Fatalf("Could not start resource: %s", err)
+		}
+
+		//Retry until Couchbase Server is online and responding
+		if err = pool.Retry(func() error {
+			var err error
+			_, err = http.Get("http://127.0.0.1:8091/pools")
+			if err != nil {
+				return err
+			}
+			return nil
+
+		}); err != nil {
+			log.Fatalf("Could not connect to docker: %s", err)
+		}
+
+		//Setup hostname
+		//xip.io hostname simply resolves to localhost. This allows us to to communicate with Couchbase on localhost
+		//to prevent the issue where the internal docker IP is used. See xip.io website for more info.
+		data := url.Values{}
+		client := &http.Client{}
+		data.Add("hostname", "127.0.0.1.xip.io")
+		req, err := http.NewRequest("POST", "http://localhost:8091/node/controller/rename", strings.NewReader(data.Encode()))
+		fatalError(`Unable to set hostname`, err)
+		req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
+		req.SetBasicAuth("Administrator", "password")
+		_, err = client.Do(req)
+		fatalError(`Unable to set hostname`, err)
+
+		//Set index storage mode
+		data = url.Values{}
+		data.Set("storageMode", "plasma")
+		req, err = http.NewRequest("POST", "http://localhost:8091/settings/indexes", strings.NewReader(data.Encode()))
+		fatalError(`Unable to set index mode`, err)
+		req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
+		req.SetBasicAuth("Administrator", "password")
+		_, err = client.Do(req)
+		fatalError(`Unable to set index mode`, err)
+
+		//Setup required services
+		data = url.Values{}
+		data.Set("services", "kv,n1ql,index,fts")
+		response, err := http.PostForm("http://localhost:8091/node/controller/setupServices", data)
+		fatalError(`Unable to setup services`, err)
+
+		//Setup memory quotas for each service
+		data = url.Values{}
+		data.Set("memoryQuota", "3096")
+		data.Set("indexMemoryQuota", "1024")
+		data.Set("ftsMemoryQuota", "1024")
+		response, err = http.PostForm("http://localhost:8091/pools/default", data)
+		fatalError(`Unable to setup services`, err)
+
+		//Set admin credentials
+		data = url.Values{}
+		data.Set("password", "password")
+		data.Set("username", "Administrator")
+		data.Set("port", "SAME")
+		response, err = http.PostForm("http://localhost:8091/settings/web", data)
+		fatalError(`Unable to set admin credentials`, err)
+
+		//Define bucket specs
+		bucketSettings := gocb.BucketSettings{
+			FlushEnabled:  true,
+			IndexReplicas: false,
+			Name:          "test_data_bucket",
```

Comment:
We should use the constants defined in `constants.go` for all of the bucket and user settings

https://github.com/couchbase/sync_gateway/blob/master/base/constants.go#L18-L38